File: solutions/java/series/1/src/main/java/Series.java
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class Series {
    private final String string;

    Series(String string) {
        if (string == null || string.isEmpty()) {
            throw new IllegalArgumentException("series cannot be empty");
        }

        this.string = string;
    }

    List<String> slices(int length) {
        if (length <= 0) {
            throw new IllegalArgumentException("slice length cannot be negative or zero");
        }

        if (length > string.length()) {
            throw new IllegalArgumentException("slice length cannot be greater than series length");
        }

        return IntStream.rangeClosed(0, string.length() - length)
                       .mapToObj(i -> string.substring(i, i + length))
                       .collect(Collectors.toList());
    }
}

File: solutions/python/series/1/series.py
def slices(series, length):
    _validate_series(series, length)
    return _get_slices(series, length)

def _validate_series(series, length):
    if not series:
        raise ValueError("series cannot be empty")

    if length < 0:
        raise ValueError("slice length cannot be negative")

    if length == 0:
        raise ValueError("slice length cannot be zero")

    if length > len(series):
        raise ValueError("slice length cannot be greater than series length")

def _get_slices(series, length):
    series_slices = []
    number_of_slices = len(series) - length + 1
    for start_index in range(number_of_slices):
        end_index = start_index + length
        series_slices.append(series[start_index:end_index])

    return series_slices